include/ofi_mem.h

```diff
@@ -306,7 +306,7 @@ struct util_buf_footer {
 };
 
 union util_buf {
-	struct slist_entry entry;
+	struct dlist_entry entry;
 	uint8_t data[0];
 };
```

Comment:
I thought the util pools maintained the user's data between insertion and free calls?  That's not the case, obviously, but what am I thinking of?  There was some change to buffer pools somewhere to allow rxm to pre-format headers.

Reply:
Maybe it was the freestack code.  And freestack isn't dynamic, so we already have an optimized version of a non-dynamic buffer pool.

Reply:
util buf pool allows pre-formatting the buffers. The users should take care that they just don't use the memory used `entry`.

Reply:
This isn't maintainable, since the change to entry broke the user.  The buffers now have a footer.  We should always use the footer and move entry there.

This brings up a point regarding the index values assigned to buffers.  If we're freeing regions, it makes sense to always free from the highest index down.